rtrlib/transport/ssh/ssh_transport.c

```diff
@@ -74,6 +75,11 @@ int tr_ssh_open(void *socket)
 
 	if (config->client_privkey_path)
 		ssh_options_set(ssh_socket->session, SSH_OPTIONS_IDENTITY, config->client_privkey_path);
+	if (config->new_socket && config->opaque_info) {
```

Comment:
I'm not sure it makes sense to require new_socket and opaque_info to be set, some might not care about any context information.